core/unix/signal.c

```diff
@@ -5722,8 +5724,8 @@ handle_sigreturn(dcontext_t *dcontext, void *ucxt_param, int style)
             memcpy(&prevset.sig[1], &frame->IF_X86_ELSE(extramask, uc.sigset_ex),
                    sizeof(prevset.sig[1]));
         }
-# endif
         set_blocked(dcontext, &prevset, true/*absolute*/);
```

Comment:
I'm a little worried that moving it inside the #else branch, we might miss calling `set_blocked` when we add an implementation for AArch64. Maybe it would make more sense to initialize `prevset.sig` just before `ASSERT_NOT_IMPLEMENTED(false); /* FIXME i#1569 */` with a comment that that silences a warning with GCC 7. That might introduce a unused assignment warning with other compiler versions though